examples/src/zengine_examples/00_Minimal/main.nim

```diff
@@ -1,98 +1,204 @@
-      # Shutdown if X button clicked
-      of QuitEvent:
-        running = false
-
-      of KeyUp:
-        let keyEvent = cast[KeyboardEventPtr](addr evt)
-        # Shutdown if ESC pressed
-        if keyEvent.keysym.sym == K_ESCAPE:
-          running = false
-
-        # Get some info about the camera state
-        if keyEvent.keysym.sym == K_C:
-            echo("camera.position=" & $camera.position)
-            echo("camera.target=" & $camera.target)
-            echo("camera.up=" & $camera.up)
-
-      # Update camera if mouse moved
-      of MouseMotion:
-        let mouseMoveEvent = cast[MouseMotionEventPtr](addr evt)
-        mouseXRel = mouseMoveEvent.xrel
-        mouseYRel = mouseMoveEvent.yrel
-
-      else:
-        discard
+    case evt.kind
+    of QuitEvent:
+      mainLoopRunning = false
+    of MouseMotion:
+      var mouseMoveEvent = cast[MouseMotionEventPtr](addr evt)
+      mouseXRel = mouseMoveEvent.xrel
+      mouseYRel = mouseMoveEvent.yrel
+    of MouseWheel:
+      var mouseWheelEvent = cast[MouseWheelEventPtr](addr evt)
+      mouseWheelMovement = mouseWheelEvent.y
+    of KeyUp:
+      # Close on ESC Pressed
+      let keyEvent = cast[KeyboardEventPtr](addr evt)
+      if keyEvent.keysym.sym == K_ESCAPE:
+        mainLoopRunning = false
+    else:
+      discard
 
-  # Update the camera's position
-  camera.update(0, -mouseXrel, -mouseYRel)
+  pollInput()
 
-  clock.tick()
+  camera.update(mouseWheelMovement, mouseXRel, mouseYRel)
+  
+  tick()
 
-  # Start drawing
   beginDrawing()
-  clearBackground(BLACK)
-
+  clearBackground(ZENGRAY)
   begin3dMode(camera)
-  drawCube(vec3f(0, 2, 0), 1, 1, 1, RED)
-  drawPlane(vec3f(0, 0, 0), vec2f(32, 32), GRAY)
+  drawPlane(vec3f(0.0, 0.0, 0.0), vec2f(32.0, 32.0), GREEN)
+  drawCube(vec3f(-16.0, 2.5, 0.0), (0.0, 0.0, 1.0, 0.0), 1.0, 5.0, 32.0, BLUE)
+  drawCube(vec3f(16.0, 2.5, 0.0), (0.0, 0.0, 1.0, 0.0), 1.0, 5.0, 32.0, RED)
+  drawCube(vec3f(0.0, 2.5, 16.0), (0.0, 0.0, 1.0, 0.0), 32.0, 5.0, 1.0, WHITE)
+  drawModel(model, WHITE, timeElapsed() * 1000)
   end3dMode()
-
-  drawText("Hello zengine!", 8, 8, 16, ZColor(r: 0xFF, g: 0xFF, b: 0xFF, a: 0xFF))
-
-  # done with drawing, display the screen
   endDrawing()
   swapBuffers()
-
-# Shutdown
-zengine.core.shutdown()
+  runGC()
+
+var initFunc : proc()
+
+var initDone = false
+proc mainLoopPreload() {.cdecl.} =
+    if initDone:
+        if mainLoopRunning:
+          mainLoopInner()
+    else:
+        let r = EM_ASM_INT """
+        return (document.readyState === 'complete') ? 1 : 0;
+        """
+        if r == 1:
+            GC_disable() # GC Should only be called close to the bottom of the stack on emscripten.
+            initFunc()
+            initFunc = nil
+            initDone = true
+
+template runApplication*(initCode: typed): stmt =
+  initFunc = proc() =
+      initCode
+  emscripten_set_main_loop(mainLoopPreload, 0, 1)
+
+runApplication:
+  startApplication()
+  
+
+# # State variables
```

Comment:
All this comment stuff looks like some leftover cruft.  Is that the case?  If so, then purge it.